server/container/util/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -38,39 +64,24 @@
         return new FluentFutureStream<>(completableFuture);
     }
 
+    public static <T, U> FluentFutureStream<U> of(Stream<CompletableFuture<T>> completableFuture,
```

Comment:
all the unboxer stuff is suspicious to me. It feels like it could all be replaced by map+flatten